src/backend/types/dns-discovery.d.ts

```diff
@@ -0,0 +1,79 @@
+// Type definitions for dns-discovery 6.2.3
```

Comment:
Did you generate these, or write them by hand? Curious what tool, if any, you used.

Reply:
I write them by hand based on the docs, but often only typing the pieces that I use. Most modules already have open-source types in [Definitely Typed](http://definitelytyped.org) and can be installed by `npm i -D @types/dns-discovery`, but some less popular modules do not have community maintained types. It's confusing learning how to write these. You can just create a placeholder file with just the line: `declare module 'dns-discovery' {}` and Typescript will basically just ignore the module and not type-check your usage of it. I find writing these types helpful for sanity-checking how I am using the API of a module.